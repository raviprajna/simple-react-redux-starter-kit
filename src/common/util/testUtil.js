import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export function getComponentTree( component){
    const wrapper = shallow(component);
    const shallowComponent = wrapper.dive();
    return toJson(shallowComponent);
}
export function getWrapper( component){
    const wrapper = shallow(component);
    return wrapper;
}