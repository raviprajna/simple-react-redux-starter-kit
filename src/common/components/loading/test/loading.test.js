import React from 'react';
import Loading from '../';
import {getComponentTree, getWrapper } from '../../../util/testUtil.js';
import { Flex, Box } from 'rebass';

describe('<Loading /> Component Test', () => {
  it('renders <Loading /> components', () => {

  const wrapper = getWrapper(<Loading />);

  // Print Html to console
  console.log(wrapper.debug());

  // Assert if loading contains particular react element.
  expect(wrapper.contains(<Box> Loading... </Box>)).toEqual(true);

  // Assert with snapshot
  expect(getComponentTree(<Loading />)).toMatchSnapshot();


  });

});