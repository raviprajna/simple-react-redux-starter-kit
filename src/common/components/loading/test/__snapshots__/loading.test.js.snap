// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> Component Test renders <Loading /> components 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          "display: flex;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
        ],
      },
      "displayName": "Styled(styled.div)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "propTypes": Object {
        "alignItems": [Function],
        "flexDirection": [Function],
        "flexWrap": [Function],
        "justifyContent": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  justifyContent="center"
>
  <styled.div>
     Loading... 
  </styled.div>
</StyledComponent>
`;
